src/sentry/api/serializers/models/external_actor.py

```diff
@@ -0,0 +1,34 @@
+from typing import Any, Mapping, Optional
+
+from sentry.api.serializers import Serializer, register
+from sentry.models import ExternalActor, User
+from sentry.types.integrations import get_provider_string
+
+
+@register(ExternalActor)
+class ExternalActorSerializer(Serializer):  # type: ignore
+    def serialize(
+        self,
+        obj: ExternalActor,
+        attrs: Mapping[str, Any],
+        user: User,
+        key: Optional[str] = None,
+        **kwargs: Any,
+    ) -> Mapping[str, Any]:
+        provider = get_provider_string(obj.provider)
+        data = {
+            "id": str(obj.id),
+            "provider": provider,
+            "externalName": obj.external_name,
+        }
+
+        if obj.external_id:
+            data["externalId"] = obj.external_id
+
+        # Extra context `key` tells the API how to resolve actor_id.
+        if key == "user":
+            data["userId"] = str(obj.actor.resolve().id)
+        elif key == "team":
+            data["teamId"] = str(obj.actor.resolve().id)
```

Comment:
`.resolve()` performs a query, so this will results in N+1 queries. Ideally we'd do this in bulk in `get_attrs`. @iProgramStuff did we ever have a way to perform these resolutions in bulk? We might just want a `resolve_bulk` method on the manager, maybe.

Reply:
👍🏽  I copied the code from `RuleSerializer`.

Reply:
Ok, I think we should probably consolidate this into `resolve_bulk` at some point, but it doesn't have to be this pr since it's pretty large. I'll let you and @iProgramStuff work out the best way to do that, but I bet this won't be the last time we need to bulk resolve some actors.